### PR TITLE
Migrate provider-aware manifest and typed compatibility blocks

### PR DESCRIPTION
**Pull request by ronaldmannak**
## Purpose

Continue the focused replacement of #55 / issue #14, stacked on the now-approved #141. Implements MVP-PLAN.md §§5 and 10 under ADR 0002/0003 without importing old sanitizer ancestry.

## Changes

- Provider-aware immutable manifest, with its own schema 2, positive revision and fixed decode/resource errors.
- Preserve all original rule selectors (host, desktop, runtime, CLI paths/capabilities, guest and provisioning identity) plus explicit runtime provider and exact Codex CLI installation count.
- Replace arbitrary blocking-reason strings with closed CompatibilityBlockReason values and Guesthouse-owned messages.
- Preserve targeted repairs and required blocked-state recovery; missing observed selectors do not match a rule.
- Retain the original nested API names as aliases to the independently tested entry types from #141.
- Bundle an explicitly unverified empty candidate manifest. Old Tart placeholders remain in the preserved #55 branch, not repurposed as Lume evidence.

## Validation

**270 Core tests / 23 suites pass**, warnings-as-errors, including the finalized #140/#141 review corrections. Tests cover schema/revision rejection, resource round trip, required identity, typed reasons, selectors, provider isolation, capability bounds, recovery completion and unknown-field non-propagation. Markdown and diff checks pass.

Own diff: **467 changed lines / 6 files**.

## Scope and merge order

#138 → #139 → #140 → #141 → this PR. The evaluator migration is committed and tested locally but will be published after this parent clears, to limit repeated downstream CI. Private manifest notes/identity/evidence references are not diagnostic attachments. No runtime probes, persistence, provider selection or hardware certification is claimed; #14 is not closed by this intermediate PR.

Wait for exact-head green Xcode Cloud and a matching completed Codex review with an original-message 👍 and no unresolved findings. The owner merges.

## Review follow-up

The missing installation-count selector (P2) is fixed in 5db4e27f with round-trip, exact-match, repaired/unknown-count, omitted-selector and malformed-input coverage. The composed evaluator also passes 284 tests locally and remains unpublished while this parent gets fresh CI and Codex approval.

### CODE REVIEW

File: Packages/GuesthouseCore/Sources/GuesthouseCore/Compatibility/CompatibilityIncompatibility.swift [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    public let codexDesktopVersion: String?
+    public let codexDesktopBuild: String?
+    public let codexDesktopPath: String?
+    public let runtimeProtocolVersion: Int?
+    public let runtimeProvider: VMProvider?
+    public let runtimeVersion: String?
+    public let guestMacOSBuild: String?
+    public let xcodeBuild: String?
+    public let codexCLIVersion: String?
+    public let codexCLIPath: String?
```
[the remark this answers is not in the document]
Comment (ronaldmannak) [reply Packages/GuesthouseCore/Sources/GuesthouseCore/Compatibility/CompatibilityIncompatibility.swift:75]: Fixed in 5db4e27f. The optional exact codexCLIInstallations selector now participates in stored identity, initialization, decoding and applies(to:). Swift Testing regressions cover zero/duplicate counts, repaired count 1, unknown observations, omitted-selector wildcard behavior, full manifest round trips and malformed selector rejection. The full parent Core suite passes 270 tests / 23 suites with warnings-as-errors. A separate evaluator integration regression is also validated locally (284 tests / 24 suites): a decoded count-2 rule blocks duplicates, stops blocking after repair to the previously verified count-1 tuple, and never verifies an unknown count. That child remains unpublished until this parent clears. No diagnostic text or provider/hardware policy changed.
